src/bucket.js

```diff
@@ -2326,15 +2327,32 @@ class Bucket extends common.ServiceObject {
     }
 
     function upload() {
-      let sourceStream;
-
       if (isURL) {
-        sourceStream = request.get(requestOptions);
-      } else {
-        sourceStream = fs.createReadStream(pathString);
+        const url = requestOptions.url;
```

Comment:
Line 2108 needs to be updated-- it shows that we accept the options that `request` does. Considering that, this will be a breaking change unless we convert all `request` options to match `node-fetch`.